core/src/main/scala/org/apache/spark/sql/rf/TileUDT.scala

```diff
@@ -59,34 +62,38 @@ class TileUDT extends UserDefinedType[Tile] {
       // TODO: review matches there
       case ref: RasterRef =>
         val ct = UTF8String.fromString(ref.cellType.toString())
-        InternalRow(ct, ref.cols, ref.rows, null, ref.toInternalRow)
+        InternalRow(ct, ref.cols, ref.rows, null, null, ref.toInternalRow)
       case ProjectedRasterTile(ref: RasterRef, _, _) =>
         val ct = UTF8String.fromString(ref.cellType.toString())
-        InternalRow(ct, ref.cols, ref.rows, null, ref.toInternalRow)
+        InternalRow(ct, ref.cols, ref.rows, null, null, ref.toInternalRow)
       case prt: ProjectedRasterTile =>
         val tile = prt.tile
         val ct = UTF8String.fromString(tile.cellType.toString())
-        InternalRow(ct, tile.cols, tile.rows, tile.toBytes(), null)
+        InternalRow(ct, tile.cols, tile.rows, tile.toBytes(), null, null)
+      case bt: BufferTile =>
+        val tile = bt.toArrayTile()
+        val ct = UTF8String.fromString(tile.cellType.toString())
+        InternalRow(ct, tile.cols, tile.rows, tile.toBytes(), bt.gridBounds.toInternalRow, null)
```

Comment:
I don't know how much that makes sense though, mb it is better not to preserve TileUDT "polymorphism" there. cc @metasim

Reply:
Tile polymorphism is really important for user ergnomics. Can all Tiles be BufferTiles?

Reply:
@metasim this change is pretty minor and keeps the PR within its scope: https://github.com/locationtech/rasterframes/pull/563/files#diff-4c1508a758314ff3b5773ada749119d0853a72490b4836749acf00b302fbc74dR59-R92 

both RasterRef and BufferTiles work as expected